internal/ldso-cache/ldsocache.go

```diff
@@ -547,23 +548,17 @@ func (hdr *LDSORawCacheHeader) Write(w io.Writer) error {
 func ParseLDSOConf(fsys fs.FS, ldsoconf string) ([]string, error) {
 	conf, err := fsys.Open(ldsoconf)
 	if err != nil {
-		Debugf("Warning: Could not open config file %s\n", ldsoconf)
+		debugf("Warning: Could not open config file %s\n", ldsoconf)
 		return nil, err
 	}
 	defer conf.Close()
-	contents, err := io.ReadAll(conf)
-	if err != nil {
-		Debugf("Warning: Could not read config file %s\n", ldsoconf)
-		return nil, err
-	}
+
+	scanner := bufio.NewScanner(conf)
 	libpaths := []string{}
 
-	lines := strings.Split(string(contents), "\n")
-	for _, line := range lines {
-		idx := strings.Index(line, "#")
-		if idx > -1 {
-			line = line[:idx]
-		}
+	for scanner.Scan() {
+		line := scanner.Text()
+		line, _, _ = strings.Cut(line, "#") // remove comments
```

Comment:
ah, i think i remember you suggesting this the first time around, sorry I missed it!